ludwig/datasets/__init__.py

```diff
@@ -80,6 +80,11 @@ def list_datasets() -> List[str]:
     return sorted(_get_dataset_configs().keys())
 
 
+def list_dataset_configs() -> Dict[str, DatasetConfig]:
```

Comment:
Could you use an accessor function or function(s) like: 

```
@PublicAPI
def tasks_for_dataset(dataset_name) -> List[str]
```

and/or 

```
@PublicAPI
def datasets_for_task(dataset_name) -> List[str]
```

or whatever we need on the backend.  Otherwise we are implicitly making `DatasetConfig` part of the public API, which is something I've tried to avoid.  Once client code is using a data structure, we must worry about what we are breaking every time we change it.

See for example `describe_dataset` in `datasets/__init__.py` - its purpose is to provide the requested information **without** exposing the underlying data structure.

Reply:
Made the change to get as close to this as I can. We need the task and the name and in order. I realize it's slightly bespoke but fundamentally we need this data so I feel like this is a lean way of doing this. Let me know what you think!